packages/core-data/src/actions.js

```diff
@@ -69,6 +69,11 @@ export function addEntities( entities ) {
  * @return {Object} Action object.
  */
 export function receiveEntityRecords( kind, name, records, query, invalidateCache = false ) {
+	// Auto drafts should not have titles, but some plugins rely on them so we can't filter this
+	// on the server.
+	records = castArray( records ).map( ( record ) =>
+		record.status === 'auto-draft' ? { ...record, title: '' } : record
+	);
```

Comment:
this bundles a post-specific behavior to a generic entity handler. Can't we instead apply the filter to the REST API endpoint only instead of a global PHP filter?

Reply:
That would still break plugins that use the REST API and rely on the current behavior.

>this bundles a post-specific behavior to a generic entity handler.

Yeah, it's unfortunate, but it won't break non-posts, it will just return the records.

Reply:
I thought the breakage was related to the usage of a specific WordPress function (and not the bahavior itself).

Is there a clean way we can move this out of the entities or make it generic somehow?

Reply:
>I thought the breakage was related to the usage of a specific WordPress function (and not the bahavior itself).

Both

>Is there a clean way we can move this out of the entities or make it generic somehow?

Not really, unless there is a way of adding plugin specific filters that I am not aware of.

Reply:
Can we at least create an issue maybe to not forget to think about how we can make this generic. Maybe some functions on the entities config or something?

Also, there's something that feels wrong here. We don't want this behavior in a generic way server-side but we do want it in the client side (which can is a bit non-sense) (The entities abstraction is generic and not specific to the editor).

Reply:
Auto draft titles don't add any value and can only confuse users in the context of the editor, so we decided to remove them.

Auto drafts should also be converted to drafts on explicit saves, because users would never want to save a post and keep it as an auto draft.

We decided to implement these 2 behaviors in a server side filter in #16814, but that broke certain plugins, so we took a different approach in #17317.

The new approach is to do what the filter was doing, but in the client's core data entity handling code.

@youknowriad Pointed out that this brings the same problem the behavior modification was producing in the server, to the client: a global modification of how the API treats auto drafts. It's a good point, but one could also argue that this is the "correct" way auto drafts should behave and since nothing is depending on the new client code yet, we can do it this way from the start.

Reply:
Like that?

Reply:
That works for me. What bothers me is that we're harding this in the entities actions, while it's specific to the post entities only.  Do you think there's any way to move this to the editor store instead?

Reply:
Yes, but I don't think we want that.

By having it here we enforce the correct behavior on all new client side code, and non-post entities are not affected.